lib/active_model/serializer/attribute.rb

```diff
@@ -0,0 +1,38 @@
+module ActiveModel
+  class Serializer
+    Attribute = Struct.new(:name, :block, :key, :condition_type, :condition) do
+      def initialize(name, block, options)
+        super(name, block)
```

Comment:
By 'work' you mean defining accessors for the options?  Ok, I get that.  I think I'd rather add an attr_accessor than change the Struct definition if I needed to change the options. It took me a couple of reads of the object before I realized you were treating most of the struct attributes as options.  (I'm not new to subclassing the initialize... I just usually use it for setting defaults for unset values, rather than as implicit accessors).

Do you foresee taking advantage of the struct's comparable properties? I'm trying to balance the clever vs. useful in my head.